app/routes/goal_routes.py

```diff
@@ -1 +1,114 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import abort, Blueprint, make_response, request, Response
+from app.models.task import Task
+from app.models.goal import Goal
+from ..db import db
+from sqlalchemy import cast, String, Integer
+from .route_utilities import validate_model
+from datetime import datetime
+import os
+import requests
+
+bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+
+@bp.post("") 
+def create_goal():  
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+
+    except KeyError as error:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+        
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
```

Comment:
Note how this route and your `POST` route for tasks are essentially the same. We could use `create_model` from flasky to D.R.Y. this up.